Publish first product update since migration - January 2024 (#199)
andela-technology/qualified-docs: content/changes.md

Learn about notable new features, fixes, and other improvements to the application!

## January 31, 2024 

### Features 
 

> - **Learnosity Partnership Integration** 

Reviewer: ```suggestion
- **Qualified and Learnosity Integration** 
```

I'm flexible on this though, either way works. The link below says "partnership" so it seems a bit repetitive.

>     [Qualified and Learnosity Partnership: Bridging the Skills Gap with Developer Assessments](https://andela.com/blog-posts/qualified-and-learnosity-partnership-bridging-the-skills-gap-with-developer-assessments)

Reviewer: I know Jake mentioned a simple note along these lines. But I still think it'd benefit from a little bit of context for what the feature is. Not a blocker or must-do, but I like my suggestion in https://github.com/andela-technology/qualified-docs/pull/199#discussion_r1575262933. Feel free to merge without it if you and Jake are confident with the current approach.

Learn about notable new features, fixes, and other improvements to the application!

> ## January, 2024 

Reviewer: I would pick a specific date here so it's in keeping with the rest of the product updates, or use "January 2024" without a comma if you do want to switch to a full month. For the backdated entries, you could use an approximate date or January 1, 2024. I think having the exact date is nice, personally, so there's a clear date of release for the features.

Author: I agree but since we are going to go to a monthly cadence I prefer to keep the monthly format and at the end of each month or at the beginning of the month about the closed month. I understand that we want to keep the format but in any case I would almost go for the **end of the month** instead of the beginning. Now it is a bit confusing because I am generating months already past. It could be January 31, 2024.

Reviewer: If you use the same day every month, that's fine with me. January 31, 2024 and so on works for me. I think this is better than "January 2024" which is a bit vague. Once we catch up, we'll want to use the exact day it's posted as the date.

Author: That is why I don't want to be attached to an specific date, I prefer at this point to go to a more monthly cadence without matter the exact date just the updates or improvements released during the last month.

### Features 
 
- **Learnosity Partnership Integration** 

    [Learnosity](https://learnosity.com/) has officially announced our partnership and integration into their platform to their custom base, via website, newsletter, and socials! 

>     This integration allows us to expand our reach into the L&D space, acting as an important add-on component within Learnosity's deep offering across many different learning domains. We've closed our first partner sale via iCEV, and already have other potential deals beginning to surface.       We will be presenting internally to their sales team later this month to further deepen our partnership and enable their sales team to bring more business to us.  

Reviewer: This note feels a bit like an internal note to our own staff. The purpose of the Product Updates is for us to communicate relevant platform changes to Qualified's non-Andela customers.

I'm not sure if the Learnosity integration would be relevant to any of our standard customers, unless they're potential customers of Learnosity--if so, maybe it's worth mentioning with a focus on how they can use it. But details about landing iCEV and us pursuing new customers seems outside the purview of Product Updates.

So I'd suggest sticking to:
- bug fixes (e.g. crashes, performance problems, etc)
- new challenges
- new platform features
- new language versions
- improvements to existing features
- general integrations that Qualified customers can use (Greenhouse, Zapier, etc)
- ...in general, technical changes

And avoiding:
- Andela-specific changes (proctoring, resume, etc)
- sales we've closed with customers or integrations that don't affect our clients in general
- marketing announcements (the product changes never mentioned we were acquired by Andela, to pick a somewhat extreme example)
- ...in general, non-technical or internal changes

Author: I think it's relevant because like the case of Thinkful who are also Learnosity customers they want to learn more about Learnosity <> Qualified integration. I think the note is promoting and putting the emphasis of a partnership we were able to establish with give us and our customer extra benefits.

Reviewer: The topic is fine to keep, but I'd suggest using the [SmartIDE launch](https://docs.qualified.io/changes/#may-11-2022) as a template, which is probably the most marketing-style announcement in the changelog.

The purpose of this page is to provide a list of technical changes to document the evolution of the application/platform itself over time, not acquisitions of new customers or internal meetings--I don't see anything like that in the rest of the changelog. We do not mention any customers by name, for example.

How about something like:

> We've partnered with [Learnosity](https://learnosity.com/) to integrate Qualified coding challenges into Learnosity assessments! Our custom question type enables students to solve project and classic coding challenges in a Learnosity assessment, powered by Qualified's [Embed](https://www.qualified.io/embedded) engine.
> 
> Visit the [release announcement](https://help.learnosity.com/hc/en-us/articles/14870489871133-2024-1-LTS-Release-Announcement) and [Learnosity's Qualified partner page](https://learnosity.com/company/partners/partners-directory/qualified/) to learn more.

This gives them all of the information they need and provides the right level of technical detail, while directing them to the main marketing-style material, and avoiding mentioning iCEV and internal meetings.

Reviewer: Maybe Learnosity's [product changelog](https://reference.learnosity.com/releaselogs) would be useful to help clarify the flavor of a typical product changelog (they didn't mention the Qualified integration in the changelog--only in the marketing release notes).

That said, other changelogs are a bit different, so it's not the only way to do it. Some release logs are a bit more "linguistic", like ours, so we don't have to be _quite_ as stark as Learnosity. The technical customer audience, purpose and tone is the same though.

    [Release Announcement](https://help.learnosity.com/hc/en-us/articles/14870489871133-2024-1-LTS-Release-Announcement)

    [Partner Page](https://learnosity.com/company/partners/partners-directory/qualified/)

### Improvements 

- **Erlang 26** 

    Erlang 26 is now supported within project challenges. 

- **Elixir 1.15** 

    Elixir 1.15 is now supported within project challenges.

- **Add deprecated language version labels** 

>     A new “deprecated” label was added on old language versions in the challenge edit and detail views, visible to teams and content managers with the purpose of providing our users with more legibility of what are the available language versions to use. For further information please check       our [documentation](https://docs.qualified.io/creating-content/challenges/upgrading-language-versions/#upgrading-language-versions) 

Reviewer: Note that we're already in the documentation, so we should be specific about which page we're linking them to.

```suggestion
    A new "deprecated" label was added to old language versions in the challenge edit and detail views. This helps content managers identify which language versions will be removed in the future. For further information and tips on upgrading your content, please check our [guide for upgrading language versions](https://docs.qualified.io/creating-content/challenges/upgrading-language-versions/#upgrading-language-versions).
```

  
- **Performance issue when loading "Portfolio" tab of report card** 

>     Users were experiencing heavy lag, especially with scrolls, once Portfolio tab was activated and the number of assessments results was exceeding 6 or more. This issue was successfully fixed.  

Reviewer: In general, it's good to share bug fixes and performance improvements like this, but in this case, the portfolio is only available for teams with signals enabled. I don't see anything else in the product updates historically related to signals, so it may be something we skip just because it doesn't affect all customers.

Author: Yes, I wasn't sure about this because could have a bad impact over the fact we were able to solved a major performance issue. I'm going to remove it.

Reviewer: I might be misunderstanding your response, but in general, it's good to be honest about solving issues--we don't want to hide bug fixes, no matter how large. Big companies like [OpenAI do postmortems for outages](https://openai.com/blog/march-20-chatgpt-outage).

So the reason not to share this is (potentially) only that it's a flagged feature that doesn't pertain to most of our customers.

   We've partnered with Learnosity to integrate Qualified coding challenges into Learnosity assessments! Our custom question type enables students to solve project and classic coding challenges in a Learnosity assessment, powered by Qualified's Embed engine. 

  Check our post [Qualified and Learnosity Partnership: Bridging the Skills Gap with Developer Assessments](https://andela.com/blog-posts/qualified-and-learnosity-partnership-bridging-the-skills-gap-with-developer-assessments) to learn more.

### Improvements 

- **Erlang 26** 

    Erlang 26 is now supported within project challenges. 

- **Elixir 1.15** 

    Elixir 1.15 is now supported within project challenges.

- **Add deprecated language version labels** 

    A new "deprecated" label was added to old language versions in the challenge edit and detail views. This helps content managers identify which language versions will be removed in the future. For further information and tips on upgrading your content, please check our [guide for upgrading language versions](https://docs.qualified.io/creating-content/challenges/upgrading-language-versions/#upgrading-language-versions).
  

> - **Performance issue when loading "Portfolio" tab of report card**

Reviewer: This should be in a [`### Bug Fixes` section, below the challenges](https://github.com/andela-technology/qualified-docs/blob/a3da79dd9ed37bcd549b9922006d817eab0147e4/scripts/import_product_releases.py#L48).

Reviewer: Technically, you could consider any bug fix as an improvement (adding challenges could also be considered an improvement, for that matter), but improvements are things that aren't bugs.

In this case, the performance made that part of the app basically unusable and wasn't intended behavior, so it's a bug.

If you're not sure, you can check the issue type, [which confirms the bug status](https://andela.atlassian.net/browse/QUAL-415).

  Check our post [Qualified and Learnosity Partnership: Bridging the Skills Gap with Developer Assessments](https://andela.com/blog-posts/qualified-and-learnosity-partnership-bridging-the-skills-gap-with-developer-assessments) to learn more.

### Improvements 

- **Erlang 26** 

    Erlang 26 is now supported within project challenges. 

- **Elixir 1.15** 

    Elixir 1.15 is now supported within project challenges.

- **Add deprecated language version labels** 

    A new "deprecated" label was added to old language versions in the challenge edit and detail views. This helps content managers identify which language versions will be removed in the future. For further information and tips on upgrading your content, please check our [guide for upgrading language versions](https://docs.qualified.io/creating-content/challenges/upgrading-language-versions/#upgrading-language-versions).
  
### Bug Fixes

- **Fixed report card "Portfolio" tab performance issue**

  :::note
  Signals customers only
  ::: 

>   Users were experiencing heavy lag, especially with scrolls, once the Portfolio tab was activated and the number of assessments results was 6 or more. This issue was successfully fixed.  

Reviewer: Please be careful with the leading whitespace here and check the deploy preview (or preview locally):

![image](https://github.com/andela-technology/qualified-docs/assets/154474514/efede544-f2ed-4f58-8596-ca9378c9837c)

```suggestion
    Users were experiencing heavy lag, especially with scrolls, once the Portfolio tab was activated and the number of assessments results was 6 or more. This issue was fixed.
```

Really minor, but also try to avoid trailing spaces. These are usually removed by pre commit autoformatting, but we don't use it on our markdown files since we have non-standard extensions.

>     Users were experiencing heavy lag, especially with scrolls, once Portfolio tab was activated and the number of assessments results was exceeding 6 or more. This issue was successfully fixed.  

Reviewer: ```suggestion
    We fixed an issue with users experiencing heavy lag in the report card Portfolio tab when many assessment results were present.
```

### Challenges

- **ASCII 7 Segment Display challenge** 

>     This challenge tests the candidate's basic array and string manipulation.  

Reviewer: ```suggestion
    This challenge tests the candidate's basic array and string manipulation skills.
```

    A new "deprecated" label was added to old language versions in the challenge edit and detail views. This helps content managers identify which language versions will be removed in the future. For further information and tips on upgrading your content, please check our [guide for upgrading language versions](https://docs.qualified.io/creating-content/challenges/upgrading-language-versions/#upgrading-language-versions).
  
- **Performance issue when loading "Portfolio" tab of report card**

  :::note

>   Signal’s customers only

Reviewer: ```suggestion
  Signals customers only
```

  Signals customers only
  ::: 

>     Users were experiencing heavy lag, especially with scrolls, once the Portfolio tab was activated and the number of assessments results was 6 or more. This issue was fixed.

Reviewer: Sorry, one more typo fix, then we're good to go:

```suggestion
    Users were experiencing heavy lag, especially with scrolls, once the Portfolio tab was activated and the number of assessment results was 6 or more. This issue was fixed.
```
